### PR TITLE
[refactor] Add "Extract Expression" action 

### PR DESCRIPTION
**Pull request by hyp**
"Extract Expression" is a sub-action of extract.

Note: I thought I will use the new function from first NFC commit, which moves `nameForExtractedVariable` into a separate file, but decided against it.

### CODE REVIEW

**Comment by hyp** [comment]
I linked visual review in rdar.

File: lib/Tooling/Refactor/Extract.cpp [hunk 1]
```diff
@@ -1437,6 +1456,54 @@ static bool isInHeader(SourceLocation Loc, const SourceManager &SM) {
...
+llvm::Expected<RefactoringResult>
+ExtractOperation::performExpressionExtraction(ASTContext &Context,
+                                              PrintingPolicy &PP) {
+  assert(isExpressionExtraction() && "Not an expression extraction");
+  std::vector<RefactoringReplacement> Replacements;
+  const Expr *E = cast<Expr>(S);
+  QualType VarType = findExpressionLexicalType(FunctionLikeParentDecl, E,
+                                               E->getType(), PP, Context);
+  StringRef VarName = "extractedExpr";
+  std::unique_ptr<RefactoringResultAssociatedSymbol> CreatedSymbol =
```
Comment (vedantk) [inline lib/Tooling/Refactor/Extract.cpp:1468]: 'auto' for the left-hand-side?

**Comment by vedantk** [comment]
The code movement checks out, and the new logic lgtm. Do you need to teach the tester tool about CXRefactor_Extract_Expression?

edit: Never mind, I see that's what RefactoringActions.def is for.

**Comment by hyp** [comment]
To https://github.com/apple/swift-clang
   09ae941c16..64fda69e35  upstream-with-swift -> upstream-with-swift

**Pull request closed without merging**